### PR TITLE
Refactor: Add new helper crate `re_log_encoding`

### PR DESCRIPTION
**Pull request by emilk**
No new code in this PR, only moved existing code around.

Sparked by https://github.com/rerun-io/rerun/pull/1771, I wanted to fix compiling `rerun` with `--no-default-features`

This PR introduces a new crate  ~`re_transport`~ `re_log_encoding` that handles:
* encoding/decoding of LogMsg as .rrd files/streams
* streaming .rrd files over HTTP
* streaming log messages to disk as .rrd files

Questions:
* Should we move `trait LogSink` from `re_sdk` to `re_log_encoding`?
* Is ~`re_transport`~ `re_log_encoding` a good name?

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: crates/re_log_types/src/lib.rs [hunk 1]
```diff
@@ -162,9 +159,8 @@ impl std::fmt::Display for ApplicationId {
 
 /// The most general log message sent from the SDK to the server.
 #[must_use]
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, PartialEq)] // `PartialEq` used for tests
```
Comment (emilk) [inline crates/re_log_types/src/lib.rs:162]: …but the test is in another crate

**Comment by jleibs** [comment]
> How about re_log_encoding?

That works for me.

File: crates/re_log_encoding/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# re_log_encoding
+
+Part of the [`rerun`](https://github.com/rerun-io/rerun) family of crates.
+
+[![Latest version](https://img.shields.io/crates/v/re_log_encoding.svg)](https://crates.io/crates/re_log_encoding)
+[![Documentation](https://docs.rs/re_log_encoding/badge.svg)](https://docs.rs/re_log_encoding)
+![MIT](https://img.shields.io/badge/license-MIT-blue.svg)
+![Apache](https://img.shields.io/badge/license-Apache-blue.svg)
+
+Helper library for transporting Rerun log messages.
```
Comment (jleibs) [inline crates/re_log_encoding/README.md:10]:
```suggestion
Helper library for encoding Rerun log messages.
```

File: crates/re_log_encoding/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,42 @@
+//! Crate that handles transporting of rerun log types.
```
Comment (jleibs) [inline crates/re_log_encoding/src/lib.rs:1]:
```suggestion
//! Crate that handles encoding of rerun log types.
```

**Pull request merged**